retrieval_service/main.py

```diff
@@ -0,0 +1,149 @@
+        except gitlab.exceptions.GitlabGetError as e:
+            logging.info(
+                f'{self.__class__.__name__}: GitLab API error retrieving user ID {event_data["user_id"]}: {e}'
+            )
+
+    def _process_project_event(self, event_data):
+        try:
+            return self.gitlab_client.projects.get(event_data["project_id"])
+        except gitlab.exceptions.GitlabGetError as e:
+            logging.info(
+                f'{self.__class__.__name__}: GitLab API error retrieving project ID {event_data["project_id"]}: {e}'
+            )
+
+    def _process_issue_event(self, event_data):
+        try:
+            project = self.gitlab_client.projects.get(
+                event_data["object_attributes"]["project_id"]
+            )
+            return project.issues.get(event_data["object_attributes"]["id"])
+        except gitlab.exceptions.GitlabGetError as e:
+            logging.info(
+                f'{self.__class__.__name__}: GitLab API error retrieving issue ID {event_data["object_attributes"]["id"]}: {e}'
+            )
+
+    def _process_issue_note_event(self, event_data):
+        try:
+            project = self.gitlab_client.projects.get(event_data["project_id"])
+            issue = project.issues.get(event_data["issue"]["id"])
+            return issue.notes.get(event_data["object_attributes"]["id"])
+        except gitlab.exceptions.GitlabGetError as e:
+            logging.info(
+                f'{self.__class__.__name__}: GitLab API error retrieving issue note ID {event_data["object_attributes"]["id"]}: {e}'
+            )
+
+    def _process_group_event(self, event_data):
+        try:
+            return self.gitlab_client.groups.get(event_data["group_id"])
+        except gitlab.exceptions.GitlabGetError as e:
+            logging.info(
+                f'{self.__class__.__name__}: GitLab API error retrieving group ID {event_data["group_id"]}: {e}'
+            )
+
+    def push_event_to_queue(self, event, data, prefix=None):
+        queue_name = f"{prefix}_{event}"
+        serialized_data = data.to_json()
+
+        try:
+            self.redis_client.lpush(queue_name, serialized_data)
+            logging.debug(f"{self.__class__.__name__}: pushed data to {queue_name}")
+        except Exception as e:
+            logging.error(f"Error pushing data to queue {queue_name}: {e}")
+
+    def run(self):
+        while True:
+            self.poll_and_process_event()
+
+            if self.testing:
+                break
+
+            sleep(1)
+
+
+def main(
+    GITLAB_URL=os.getenv("GITLAB_URL"),
+    GITLAB_ACCESS_TOKEN=os.getenv("GITLAB_ACCESS_TOKEN"),
+    redis_conn=None,
+    testing=False,
+):
+    processor = GitlabRetrievalProcessor(
+        GITLAB_URL, GITLAB_ACCESS_TOKEN, redis_conn=redis_conn, testing=testing
+    )
+    processor.run()
+
+
+if __name__ == "__main__":
+    main(
+        GITLAB_URL=os.getenv("GITLAB_URL"),
+        GITLAB_ACCESS_TOKEN=os.getenv("GITLAB_ACCESS_TOKEN"),
```

Comment:
**nitpick (if-minor):** This is already the default on lines 133/134, no need to set it here